spec/controllers/settings/dex_connector_ldaps_controller_spec.rb

```diff
@@ -0,0 +1,201 @@
+                                            group_base_dn:      "dc=example,dc=org",
+                                            group_filter:       "(objectClass=group)",
+                                            group_attr_user:    "uid",
+                                            group_attr_group:   "member",
+                                            group_attr_name:    "name" }
+
+      end
+      it "saves the corrent name" do
+        expect(dex_connector_ldap.name).to eq("ldap1")
+      end
+      it "saves the correct host" do
+        expect(dex_connector_ldap.host).to eq("test.com")
+      end
+      it "saves the correct port" do
+        expect(dex_connector_ldap.port).to eq(389)
+      end
+      it "saves the correct start_tls value" do
+        expect(dex_connector_ldap.start_tls).to eq(false)
+      end
+      it "saves the correct certificate" do
+        expect(dex_connector_ldap.certificate.certificate).to eq("cert")
+      end
+      it "saves the correct bind_anon value" do
+        expect(dex_connector_ldap.bind_anon).to eq(false)
+      end
+      it "saves the correct bind_dn" do
+        expect(dex_connector_ldap.bind_dn).to eq("cn=admin,dc=example,dc=org")
+      end
+      it "saves the correct bind_pw" do
+        expect(dex_connector_ldap.bind_pw).to eq("admin")
+      end
+      it "saves the correct username_prompt" do
+        expect(dex_connector_ldap.username_prompt).to eq("Username")
+      end
+      it "saves the correct user_base_dn" do
+        expect(dex_connector_ldap.user_base_dn).to eq("dc=example,dc=org")
+      end
+      it "saves the correct user_filter" do
+        expect(dex_connector_ldap.user_filter).to eq("(objectClass=person)")
+      end
+      it "saves the correct user_attr_username" do
+        expect(dex_connector_ldap.user_attr_username).to eq("uid")
+      end
+      it "saves the correct user_attr_id" do
+        expect(dex_connector_ldap.user_attr_id).to eq("uid")
+      end
+      it "saves the correct user_attr_email" do
+        expect(dex_connector_ldap.user_attr_email).to eq("mail")
+      end
+      it "saves the correct user_attr_name" do
+        expect(dex_connector_ldap.user_attr_name).to eq("fn")
+      end
+      it "saves the correct group_base_dn" do
+        expect(dex_connector_ldap.group_base_dn).to eq("dc=example,dc=org")
+      end
+      it "saves the correct group_filter" do
+        expect(dex_connector_ldap.group_filter).to eq("(objectClass=group)")
+      end
+      it "saves the correct group_attr_user" do
+        expect(dex_connector_ldap.group_attr_user).to eq("uid")
+      end
+      it "saves the correct group_attr_group" do
+        expect(dex_connector_ldap.group_attr_group).to eq("member")
+      end
+      it "saves the correct group_attr_name" do
+        expect(dex_connector_ldap.group_attr_name).to eq("name")
+      end
+    end
+  end
+
+  describe "PATCH #update" do
+    let!(:certificate) { create(:certificate, certificate: "C1") }
```

Comment:
After the addition of certificates validation, we can't put anything on the `certificate` attribute. It has to be a valid certificate text. For the purpose of this test, you can simply ignore the `certificate` attribute and the factory would generate a valid one.

Also remember to